mona_sdk/client.py

```diff
@@ -596,30 +596,16 @@ def upload_config(self, config, commit_message, author=None):
             "user_id": self._user_id,
         }
 
-        try:
-            upload_response = requests.post(
-                f"{self._app_server_url}/upload_config",
-                headers=get_basic_auth_header(
-                    self.api_key, self.should_use_authentication
-                ),
-                json=config_to_upload,
-            )
-            response_data = upload_response.json()["response_data"]
-            upload_output["new_config_id"] = response_data["new_config_id"]
-            upload_output["success"] = upload_response.ok
-
-            if not upload_output["success"]:
-                # Raise an exception if asked to.
-                self._handle_service_error(UPLOAD_CONFIG_ERROR_MESSAGE)
+        upload_response = self._app_server_request("upload_config", config_to_upload)
 
-        except ConnectionError:
-            # Raise an exception if asked to.
-            self._handle_service_error(APP_SERVER_CONNECTION_ERROR_MESSAGE)
-        except JSONDecodeError:
-            # Raise an exception if asked to.
-            self._handle_service_error(UPLOAD_CONFIG_ERROR_MESSAGE)
-
-        return upload_output
+        return (
+            {
+                "new_config_id": upload_response["response_data"]["new_config_id"],
```

Comment:
if you already touch this please make it a bit more safe with 
upload_response.get("respone_data", {}).get("new_config_id")

Reply:
I changed it a bit.. added a line that changes from False to {} to solve ide problem with the fact it had 2 different returned types on the case of failure and success.